### PR TITLE
Enhancement/database table prefix (implement #10)

### PR DESCRIPTION
**Pull request by Gounlaf**

### CODE REVIEW

**Comment by Gounlaf** [comment]
@blackandred I don't know why it fails; I see some `declare(strict_types=1);` (not in all files); is it related to that?

**Comment by blackandred** [comment]
I would encourage you to add unit tests for it.

1. It's allowing to keep the good quality of code
2. Allows to be more bug-free
3. It's a good practice to test everything, so the coverage would be high and next code changes more safe

I don't know why the build is failing on PHP 7.1, I will take a look at it closer.

**Comment by Gounlaf** [comment]
I totaly agree with you about unit tests. I just don't know what to test... =/

After merging master onto my branch, build succeed (but I still don't know why exactly)

File: src/Provider/DoctrineTablePrefixServiceProvider.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?php
```
Comment (blackandred) [inline src/Provider/DoctrineTablePrefixServiceProvider.php]: We could be more consistent about the new line between <?php and namespace :smile: 
And also I suggest to put the strict typing, it's helpful very often :wink:

**Comment by blackandred** [comment]
Ok, I looked deeply into the code. You could at least test if the provider does not thrown an exception when it is first time fetched from container and built.

The `loadClassMetadata` method could be tested possibly by creating a proper `LoadClassMetadataEventArgs` object that would met all conditions or mock it and make return values the code expects in the `loadClassMetadata` method :smile: 

It is ok if you are new to testing and you have no idea, I can help you with that :wink:

**Comment by Gounlaf** [comment]
> Ok, I looked deeply into the code. You could at least test if the provider does not thrown an exception when it is first time fetched from container and built.

I will try to do it.

> The `loadClassMetadata` method could be tested possibly by creating a proper `LoadClassMetadataEventArgs` object that would met all conditions or mock it and make return values the code expects in the `loadClassMetadata` method 😄

Well... If you can do it, it would be better.

**Comment by blackandred** [comment]
Ok, I can add that test :stuck_out_tongue:

**Comment by blackandred** [comment]
I will implement the tests and create a new PR to merge it, so you can see the changes.

**Comment by blackandred** [comment]
I will at first merge this PR into a local branch, and then create a PR from local branch again.

**Comment by blackandred** [comment]
Continuing in https://github.com/Wolnosciowiec/file-repository/pull/14 , please take a look :smile:

**Comment by Gounlaf** [comment]
@blackandred: about testing the provider:

I take a look deeper; `dbs.event_manager` is another `Pimple\Container`;
I take a look in some vendor TestCase related, and I'm completely loosed :)

What I did so far (hooray \o/):

```php
<?php declare(strict_types=1);

namespace Tests\Provider;

use Tests\WolnosciowiecTestCase;

class DoctrineTablePrefixServiceProviderTest extends WolnosciowiecTestCase
{
    public function testDbsEventManagerIsPimpleContainer()
    {
        $this->assertInstanceOf('Pimple\Container', $this->app['dbs.event_manager']);
    }
}
```

**Pull request merged**